File: frontend/types/mdt.ts
export type MDTStatus = "OPEN" | "CONCLUDED";

export interface MDTCase {
  id: number;
  patient_id: number;
  submitted_by_user_id: number;
  history_summary: string;
  discussion_status: MDTStatus;
  final_recommendation?: string;
  created_at: string;
  updated_at?: string;
  submitted_by_name?: string;
  patient_name?: string;
  comment_count?: number;
}

export interface MDTCaseDetail extends MDTCase {
  comments: MDTComment[];
}

export interface MDTComment {
  id: number;
  case_id: number;
  user_id: number;
  comment_text: string;
  created_at: string;
  author_name?: string;
  author_role?: string;
}

export type RegistryType =
  | "HEPATITIS_B"
  | "LIVER_CIRRHOSIS"
  | "ENDOSCOPY"
  | "UPPER_GI_BLEEDING"
  | "COLORECTAL_CANCER";

export interface PatientRegistry {
  id: number;
  patient_id: number;
  registry_type: RegistryType;
  source_table: string;
  source_id: number;
  tagged_at: string;
  patient_name?: string;
}

File: backend/app/schemas/mdt.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.models.mdt import MDTStatus


class MDTCaseCreate(BaseModel):
    patient_id: int
    history_summary: str


class MDTCaseOut(BaseModel):
    id: int
    patient_id: int
    submitted_by_user_id: int
    history_summary: str
    discussion_status: MDTStatus
    final_recommendation: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    submitted_by_name: Optional[str] = None
    patient_name: Optional[str] = None
    comment_count: Optional[int] = None

    class Config:
        from_attributes = True


class MDTCaseDetail(MDTCaseOut):
    comments: List["MDTCommentOut"] = []


class MDTCaseConclude(BaseModel):
    final_recommendation: str


class MDTCommentCreate(BaseModel):
    comment_text: str


class MDTCommentOut(BaseModel):
    id: int
    case_id: int
    user_id: int
    comment_text: str
    created_at: datetime
    author_name: Optional[str] = None
    author_role: Optional[str] = None

    class Config:
        from_attributes = True


MDTCaseDetail.model_rebuild()
